articles/understanding-tcp-ip-transport-layer-protocols/index.md

```diff
@@ -0,0 +1,164 @@
+# Understanding TCP/IP Transport Layer (Layer 4) protocols: TCP and UDP  
+
+![Hero image](/engineering-education/understanding-tcp-ip-transport-layer-protocols/hero-image.png)
+![Source](https://en.wikiversity.org/wiki/Internet_Protocol_Analysis/Transport_Layer#/media/File:Internet_Protocol_Analysis_-_Transport_Layer.png)
+
+### Introduction
+
+Each TCP/IP application typically choose to establish a temporary communication session to deliver data between two devices using either Transport Control Protocol (TCP) or User Datagram Protocol (UDP) based on the application requirements; this is done irrespective  regard to the destination host type, the type of media over which the data is to be sent, and the pathway taken by the information [Cisco](https://www.flashcardmachine.com/cisco-1-chapter4.html)
+TCP provides a wide range of services such as error recovery, windowing, and order segment delivery to applications, whereas UDP does not.
+In this article, we intend to look at these two transport layer protocols in detail.  
+
+### Prerequisites
+
+To better understand this article, foundational knowledge of TCP/IP  Layer 1 to 3 is required.  
+
+**Content**
+  -Transport Layer responsibilities  
+  -Multiplexing
+  -Reliability of the transport layer
+  -Transport Control Protocol (TCP)
+  -User Datagram Protocol (UDP)
+  -Multiple conversations
+  -Port number
+  -Socket and Socket pairs
+  -Conclusion
+
+### Transport Layer responsibility
```

Comment:
Before starting with this section, add another section "Transport layer" to explain transport layer in detail and how it works.